src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1789,7 +1810,8 @@
               (map->NullExpression {:column cinfo :null? value}))
 
             [["~" column-name value]]
-            (let [cinfo (get-in query-rec [:projections column-name])]
+            (let [colname (first (str/split column-name #"\."))
```

Comment:
the first value is the actual column. the rest are split on the dot and turned into a -> separated list of placeholders to use in a jsonb query. This might make it more apparent:
https://github.com/puppetlabs/puppetdb/pull/2218/files#diff-c3090734eca858a02839292b86db7464R221